### PR TITLE
Add state based template select example

### PR DESCRIPTION
**Pull request by emontnemery**
## Proposed change

Add state based template select example

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [ ] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [ ] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/template.markdown [hunk 1]
```diff
@@ -592,6 +592,25 @@ template:
...
+
+This show how a state based template select can be used to call a service.
+
+```yaml
+template:
+  select:
+    - name: "Porch Camera Day-Night Mode"
+      unique_id: porch_camera_day_night_mode
+      state: "{{ state_attr('camera.porch_camera_sd', 'day_night_mode') }}"
+      options: "{{ ['off', 'on', 'auto'] }}"
```
Comment (frenck) [inline source/_integrations/template.markdown:608]: Do we really require a template here? Like this doesn't work? 🤔 

```suggestion
      options: 
        - "off"
        - "on"
        - "auto"
```

Comment (emontnemery) [reply source/_integrations/template.markdown:608]: Right, that doesn't work, hence the example.

**Review by frenck — DISMISSED** [review]

File: source/_integrations/template.markdown [hunk 2]
```diff
@@ -592,6 +592,25 @@ template:
 
 {% endraw %}
 
+### State based select - Control Day/Night mode of a camera
+
+This show how a state based template select can be used to call a service.
```
Comment (frenck) [inline source/_integrations/template.markdown:597]:
```suggestion
This show how a state based template select can be used to call a service.

{% raw %}

```

File: source/_integrations/template.markdown [hunk 3]
```diff
@@ -592,6 +592,25 @@ template:
...
+      state: "{{ state_attr('camera.porch_camera_sd', 'day_night_mode') }}"
+      options: "{{ ['off', 'on', 'auto'] }}"
+      select_option:
+        - service: tapo_control.set_day_night_mode
+          data:
+            day_night_mode: "{{ option }}"
+          target:
+            entity_id: camera.porch_camera_sd
+```
+
```
Comment (frenck) [inline source/_integrations/template.markdown:616]:
```suggestion

{% endraw %}

```

**Review by frenck — APPROVED** [review]

**Pull request merged**